src/session_concurrency/payload_dataset_manager.cc

```diff
@@ -130,9 +136,23 @@ PayloadDatasetManager::CreateSessionIdToPayloadsMap() const
 std::string
 PayloadDatasetManager::GetSessionID(size_t dataset_index) const
 {
-  const auto payload{GetPayload(dataset_index)};
+  TensorData session_id_tensor_data{};
+
+  const auto error{data_loader_->GetInputData(
+      (*parser_->Inputs())["session_id"], 0, dataset_index,
+      session_id_tensor_data)};
+
+  if (!error.IsOk()) {
+    throw std::runtime_error(error.Message());
+  }
+
+  const uint8_t* session_id_buffer{session_id_tensor_data.data_ptr};
+  const size_t session_id_byte_size{session_id_tensor_data.batch1_size};
+
+  const std::string session_id(
+      reinterpret_cast<const char*>(session_id_buffer), session_id_byte_size);
 
-  return PayloadJsonUtils::GetSessionID(payload);
+  return session_id;
```

Comment:
This is the new code that gets the session ID as a model input rather than how it was before where it as a JSON sub-field of the payload model input.